code/go/pkg/validator/validator_test.go

```diff
@@ -432,6 +433,88 @@ func TestValidateWarnings(t *testing.T) {
 	}
 }
 
+func TestValidateExternalFieldsWithoutDevFolder(t *testing.T) {
+	pkgName := "bad_external_without_dev_build"
+	tests := []struct {
+		title               string
+		invalidPkgFilePath  string
+		buildFileContents   string
+		expectedErrContains []string
+	}{
+		{
+			"valid definition",
+			"data_stream/foo/fields/ecs.yml",
+			"dependencies:\n  ecs:\n    reference: git@v8.6.0\n",
+			[]string{},
+		},
+		{
+			"build file not exist",
+			"data_stream/foo/fields/ecs.yml",
+			"",
+			[]string{
+				"field container.id with external key defined (\"ecs\") but no _dev/build/build.yml found",
+			},
+		},
+		{
+			"ecs not defined",
+			"data_stream/foo/fields/ecs.yml",
+			"dependencies: {}\n",
+			[]string{
+				"field container.id with external key defined (\"ecs\") but no definition found for it (_dev/build/build.yml)",
+			},
+		},
+	}
+
+	pkgRootPath := filepath.Join("..", "..", "..", "..", "test", "packages", pkgName)
+	devFolderPath := filepath.Join(pkgRootPath, "_dev")
+	buildFolderPath := filepath.Join(devFolderPath, "build")
+	buildFilePath := filepath.Join(buildFolderPath, "build.yml")
+
+	defer func() {
+		os.RemoveAll(devFolderPath)
+	}()
```

Comment:
Nit. It is a bit of an anti-pattern to write and remove files from the working copy. Could we do this in a temporary directory? Even if we also have to copy the rest of the package.

Reply:
Sure, changed. Copied to a temporal directory and from there the tests are performed.